proposals/002-text-utf-default.md

```diff
@@ -0,0 +1,156 @@
+This proposal is for the migration of the `text` package from its
+default UTF-16 encoding to UTF-8. The lack of UTF-8 as a default in the
+`text` package is a pain point raised by the Haskell Community and
+industry partners for many years. The Haskell Foundation is uniquely
+positioned to effect a change like this, granted there is an appetite
+for breakage, and that such a project is well-socialized and planned.
+During the meetings of the Haskell Foundation Tech Agenda Track, we
+identified the pros and cons of such a migration against historical
+attempts, and agreed that it was within our power to do.
+
+Further, we solicited both community and industry feedback to gauge the
+appetite for breakage, and what stakeholders would be affected by the
+changes. Andrew Lelechenko has offered to lead the implementation of
+this project.
+
+# Motivation
+
+-   UTF-16 by default requires that all Text values pay a premium for serialization. Arguably, the performance impact of Text is flipped
+    upside-down: most text is UTF-8, and Haskell devs pay an undue cost when working with the wrong default.
+
+-   UTF-8 is the industry standard and by far the most common text encoding, with roughly 97% of web pages existing in UTF-8. The
+    existing UTF-16 default imposes an additional hurdle to working with the vast majority of web content on earth.
+
+-   Many systems in Haskell are UTF-8 by default (e.g. Haddock)
```

Comment:
How do such systems have this default? Do these systems avoid `text` entirely? Is there an alternative to `text` that uses UTF-8?

Reply:
text using utf 16,is the IN memory rep, not the choice of encoding it reads from the world/writes back out. the text package provides translation encoders/decoders

Reply:
Perhaps it could be clarified what "are UTF-8" means? Does it mean having a UTF-8 in-memory representation by other means than the text package, or that they use text's current UTF-16 representation, but usually input/output UTF-8?

Reply:
Thanks @cartazio for the explanation. Yes, that makes sense. So this point is saying that these tools require conversion on (de)serialization, implying that the status quo for `text` makes these tools less performant. It might good to spell that out.

Reply:
Thanks all - I'll clarify what we mean here.